app/views/play/cinematic/CinematicLankBoss.js

```diff
@@ -17,164 +37,289 @@ const Lank = require('lib/surface/Lank')
+      // normalize parameters
+      this.lanks[key].thang.pos.x = pos.x !== undefined ? pos.x : this.lanks[key].thang.pos.x
+      this.lanks[key].thang.pos.y = pos.y !== undefined ? pos.y : this.lanks[key].thang.pos.y
 
-    // Ensures lank is rendered.
-    this[side].thang.stateChanged = true
+      // Ensures lank is rendered.
+      this.lanks[key].thang.stateChanged = true
+    })
   }
 
   /**
    * Returns a command that will move the lank.
-   * @param {'left'|'right'} side
+   * @param {string} key
    * @param {{x, y}} pos
    * @param {number} ms
    */
-  moveLankCommand (side, pos = {}, ms = 1000) {
-    assertSide(side)
-
-    // normalize parameters
-    pos.x = pos.x || this[side].thang.pos.x
-    pos.y = pos.y || this[side].thang.pos.y
-    if (this[side].thang.pos.x === pos.x && this[side].thang.pos.y === pos.y) {
-      return new Noop()
-    }
-
-    const animation = anime({
-      targets: this[side].thang.pos,
-      x: pos.x,
-      y: pos.y,
-      duration: ms,
-      autoplay: false,
-      easing: 'easeInOutQuart',
-      // Inform update engine to rerender thang at new position.
-      update: () => { this[side].thang.stateChanged = true },
-      complete: () => { this[side].thang.stateChanged = true }
-    })
+  moveLankCommand ({ key, resource, pos, ms = 1000, thang }) {
+    return new MoveLank(() => {
+      if (resource) {
+        this.addLank(key, this.loader.getThangType(resource))
+      }
+      pos = pos || {}
+      // normalize parameters
+      console.log('have the key', key)
+      pos.x = pos.x !== undefined ? pos.x : this.lanks[key].thang.pos.x
+      pos.y = pos.y !== undefined ? pos.y : this.lanks[key].thang.pos.y
+      if (this.lanks[key].thang.pos.x === pos.x && this.lanks[key].thang.pos.y === pos.y) {
+        return new Noop()
+      }
+      const lankStateChanged = () => { this.lanks[key].thang.stateChanged = true }
+      const animation = anime({
+        targets: this.lanks[key].thang.pos,
+        x: pos.x,
+        y: pos.y,
+        duration: ms,
+        autoplay: false,
+        delay: 500, // Hack to provide some time for lank to load.
+        easing: 'easeInOutQuart',
+        // Inform update engine to rerender thang at new position.
+        update: lankStateChanged,
+        complete: lankStateChanged
+      })
 
-    return new AnimeCommand(animation)
+      return {
+        lankStateChanged,
+        animation,
+        run: () => new Promise((resolve, reject) => {
+          animation.play()
+          animation.complete = resolve
+        })
+      }
+    })
   }
 
-  queueAction (side, action) {
-    assertSide(side)
-    this[side].queueAction(action)
-    return Promise.resolve(null)
+  /**
+   * Sets the background.
+   *
+   * Handles backgrounds that may already exist or are simply being moved.
+   * @param {Object} background Background object.
+   */
+  setBackgroundCommand ({ slug, thang: { scaleX, scaleY, pos: { x, y } } }) {
```

Comment:
Could this and `moveLank` be consolidated into a single command that instantly moves a Lank  around?

Maybe add a comment to suggest that this would be a  worthwhile refactor.

Reply:
The arguments could be the lank key, and the new  desired positions. The thang on the lank could be compared instead of the hard coded background that is currently being used.

Reply:
There definitely seems to be a common move sequence here.  Also seems ok to put off refactor.